samples/bidirectional-chat/README.md

```diff
@@ -0,0 +1,115 @@
+
+        # Create an Azure storage account in the resource group.
+        az storage account create \
+        --name $storageName \
+        --location $region \
+        --resource-group myResourceGroup \
+        --sku Standard_LRS
+
+        # Create a serverless function app in the resource group.
+        az functionapp create \
+        --name $functionAppName \
+        --storage-account $storageName \
+        --consumption-plan-location $region \
+        --resource-group myResourceGroup \
+        --functions-version 3
+        ```
+
+    3. Renaming `local.settings.sample.json` to `local.settings.json`
+    4. Publish the sample to the Azure Function you created before.
+
+        ```bash
+        cd <root>/bidirectional-chat/csharp
+        // If prompted function app version, use --force
+        func azure functionapp publish <function-app-name>
+        ```
+
+2. Update application settings
+
+    ```bash
+    az functionapp config appsettings set --resource-group <resource_group_name> --name <function_name> --setting AzureSignalRConnectionString="<signalr_connection_string>"
+    ```
+
+3. Update Azure SignalR upstream settings
+
+    Open the Azure Portal and nevigate to the Function App created before. Find `signalr_extension` key in the App keys blade.
+
+    ![Overview with auth](getkeys.png)
+
+    Copy the `signalr_extensions` value and use `az resource` command to set the upstream setting
+
+    ```bash
+    az resource update --ids <signalr-resource-id> --set properties.upstream.templates="[{'UrlTemplate': '<function-url>/runtime/webhooks/signalr?code=<signalr_extension-key>', 'EventPattern': '*', 'HubPattern': '*', 'CategoryPattern': '*'}]"
+    ```
+
+### Use a chat sample website to test end to end
+
+1. Enable function app cross origin resource sharing (CORS)
+
+    Although there is a CORS setting in local.settings.json, it is not propagated to the function app in Azure. You need to set it separately.
+
+    1. Open the function app in the Azure portal.
+    2. In the left blade, select CORS blade.
+    3. In the Allowed origins section, add `http://127.0.0.1:5500` (It is the local web server's url).
+    4. In order for the SignalR JavaScript SDK call your function app from a browser, support for credentials in CORS must be enabled. Select the "Enable Access-Control-Allow-Credentials" checkbox.
+    5. Click Save to persist the CORS settings.
+    ![CORS](cors.png)
+
+2. Install [Live Server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) for your VS Code, that can serve web pages locally
+3. Open `bidirectional-chat/content/index.html` and edit base url
+
+    ```js
+    window.apiBaseUrl = '<function-app-url>';
+    ```
+
+4. With **index.html** open, start Live Server by opening the VS Code command palette (F1) and selecting **Live Server: Open with Live Server**. Live Server will open the application in a browser.
+
+5. Try send messages by entering them into the main chat box.
```

Comment:
Maybe add a snapshot about the chatroom sample view in the front or end is more intuitive.